### DISCUSSION TITLE
Cow raising and training

### DISCUSSION THREAD

**Discussion by jeremyckahn** [jeremyckahn/farmhand#182] [Ideas]
I think that cow raising could be expanded quite a bit in the game. I’m still thinking through how this all might work, but broadly speaking I’d like Farmhand to have a deeper concept of cow raising, with experience points and “trainable” stats. The stats should ultimately be applicable to something meaningful, though I haven’t totally figured out what that might be yet. This discussion can serve as a place to refine and document this concept.

## Experience points

Cows already have a game-day-based age and interaction-based happiness level, but I think we should add a real-time-based experience point system. I’m thinking that every few minutes or so, a cow would gain an experience point. The idea is that players are incentivized to leave the game open, even if just idly. 

## Training

Cows would have various stats that can be trained. Example might include Productivity (for producing milk or fertilizer), Luck (could cause higher crop yields), and Charity (could result in random item “gifts” to the player). I’d love to get some ideas from others here. 

Cows would be created with a random spread of stats, with each stat being trainable to some independently-determined ceiling. Stats would get trained by the player choosing where to invest experience (like Mario RPG).

## Inheritance

When breeding cows, stats would be randomly passed down from parent to child. Only randomly-selected stats would be passed down, and only one from each parent (like Pokémon).

--------------------

This is all still a nascent concept and I’m refining it as ideas come to me. Feel free to share your thoughts and I’ll update this comment as the concept evolves!

**Comment by jeremyckahn** [comment]
[Per Umi on Discord](https://discord.com/channels/714539345050075176/714539345637408793/882982145725988974), this would be a good game to take inspiration for mechanics from: https://jayseesee.github.io/Crittermound-Fork/

**Comment by lstebner** [comment]
I agree it would be cool to add some more stuff to do with the cows, and I like a lot of the ideas here.

re: real time experience... i could see this working out well. you could put a cow in a "training pen" and it stays there until taken out. in the training pen you can set the stat to train, and it slowly gains experience in that stat during that time. the training pen concept fits nicely into the farmhand tiered purchase model too. the first one could train just 1 cow, then it could be upgraded to 2, all the way up to 5, and maybe then a very expensive purchase to be able to train 2 stats at a time.

thinking about the perks is a bit more challenging. Productivity definitely seems like a good one, but I wonder if we can make "milk production" and "fertilizer strength" two different categories. stronger fertilizer could lead to the higher crop yields. Luck could then be used for something rain related I was thinking. Since cows notoriously "predict" the rain, maybe luck can bring more rainy days but without chance of storm, or maybe predict a storm a day before so the player can pick up their scarecrow so it doesn't get destroyed. Charity reminds me of a stardew valley mechanic where the pigs dig up truffles, but maybe the cows in farmhand dig up ores? training this stat increases the rate.

one other idea that came to mind is that maybe the training pen requires a crop, or an ore to be placed in it which determines which stat is trained. maybe it varies randomly for every cow, or each color has a favorite, so there is variety and a light layer of strategy involved.

I will see if I can think up any different stat categories/perks.

**Reply by jeremyckahn** [reply]
I'm glad you're into the idea!

I really like the idea of special training pens. It reminds me of how you can target specific stats to train in Pokémon, which is a really cool mechanic.

>Charity reminds me of a stardew valley mechanic where the pigs dig up truffles, but maybe the cows in farmhand dig up ores? training this stat increases the rate.

💯

>one other idea that came to mind is that maybe the training pen requires a crop, or an ore to be placed in it which determines which stat is trained.

This could be really cool. We could also have it where providing more crops results in a higher stat gain!

>maybe it varies randomly for every cow, or each color has a favorite, so there is variety and a light layer of strategy involved.

I like this too. Individual cows having a random affinity for crop also reminds me of Pokémon, which is a major inspiration for this feature. :)

**Comment by jmmullins** [comment]
On my initial read through, I took training in the "behavior" sense like puppies; my mind immediately went to young/untrained cows refusing to return to the barn at night, or causing chaos on the farm by digging up crops, breaking milk bottles, or knocking over the scarecrow 😂 I fully recognize that is NOT what you meant but it was funny to think about. 

> real-time-based experience point system

Would this factor in to the existing system that determines a cow's value, or would it essentially replace the age + happiness calculations (or whatever logic you're using now)? 

> various stats that can be trained

So I actually have a good friend who raises and cares for livestock, so I had a quick chat with her about some things cows actually do around a farm (helpful, and not) and she brought up a couple interesting things. One was tilling/plowing the fields (could tie this to Strength training, or just actually teach Tilling as a skill); in the game's current form I know there's not quite a need for it, but it was such an "oh, duh" moment that I had to mention it, as it seems highly applicable to the primary function of the game (i.e. the farming). 

Tying in with the above (if you go with Strength for the skill), she mentioned generic moving of heavy thing, even though we don't have much in the way of needing to "move" things or "building" anything with heavy materials. One thought I had around this is that maybe instead of buying the combine harvester, you could train your cows to help with harvest but there's a X% chance a cow will eat/trample some of your crops during the harvest.

On the "just for fun" side, I also see a possibility of something like dressage for cows, (Showmanship, or Study of Mooovement) which wouldn't really have any benefits other than the pure joy of getting to see that cow hop around in a different, silly way (alternate movement pattern from the untrained cows). I will say I have had the game open in the background while working recently, and seeing little cow feet hopping around the tiny portion of the window that was visible made my day better. I definitely sat there and watched them at a few points; the current movement animations are simple but adorable, so anything that makes them even cuter / more fun to watch would be a fun idle incentive. 

A skill my friend actually brought up first is in NO WAY useful to the current concept of the game because the player doesn't have an in-game avatar that has to move around a la The Sims or some of the other farming games, HOWEVER.. she brought up the fact that she was able to ride her favorite cow like a big slow horse (which I have seen and is hilariously great). I respect the lack of having to move an avatar around the screen (intentional user slow down is a shit move), but if in some distant future you wanted introduce _the option_ to have a Sims avatar-type concept walking around your farm (which I know requires more graphics), please consider a skill like Ramble which lets the player ride a cow around the field for a small advantage (probably not speed though 🤣).

My absolute FAVORITE skill that she mentioned is how much they love playing with (pushing / "kicking") balls! Yet another "just for fun" skill, you could train them to play soccer in the yard (e.g. they "kick" a ball, then follow where it went; possibly could use some of the current logic for where the cows move and just have the ball move along the same path but at differing timing). Or implement soccer-playing as just a fun thing if you buy them a ball in Supplies and assign it to them (like the hug machines).  

I recognize not much of this is "useful" to gameplay, but could definitely be fun! You may think I'm done rambling about cows, but this wasn't even one of the topics I had on my list, so off to start more cow threads.

**Reply by jeremyckahn** [reply]
@jmmullins I love all these ideas! Thanks for sharing them. I think a lot of these would fit well with Farmhand!

> Would this factor in to the existing system that determines a cow's value, or would it essentially replace the age + happiness calculations (or whatever logic you're using now)?

I think it would make sense to combine this with the cow value determination logic rather than replace it. Idle games aren't for everyone, so I don't want to lean too heavily on those mechanics. I'm thinking that it would be a nice addition for people who do like to leave games running in the background.

> if in some distant future you wanted introduce the option to have a Sims avatar-type concept walking around your farm (which I know requires more graphics), please consider a skill like Ramble which lets the player ride a cow around the field for a small advantage (probably not speed though 🤣).

Regarding avatars, it was a conscious design decision to leave avatars out of the game. I want all players to feel like included when playing Farmhand, but that becomes difficult to achieve with an avatar because it has inherent characteristics such as gender or race (or it's so abstract that it adds nothing to the experience). Games like Stardew Valley manage this well by having a robust avatar creation system. I'm not inherently against having such a system in Farmhand, it would just require a lot of art assets and careful consideration to be fully inclusive. I like the Ramble idea, though! 🙂
